pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -46,18 +46,18 @@ type BuildRunSpec struct {
 	// +optional
 	Output *Image `json:"output,omitempty"`
 
-	// Used for canceling a buildrun (and maybe more later on).
+	// StateOverride is used for canceling a buildrun (and maybe more later on).
 	// +optional
-	Status BuildRunSpecStatus `json:"status,omitempty"`
+	StateOverride BuildRunStateOverride `json:"stateOverride,omitempty"`
```

Comment:
Sorry I am later to the party. I don't think that `StateOverride` makes sense. Since the `spec` is really the requested state of whatever, and `status` is the current/actual state, we should probably read the file names as such. So if you made this field `Status`, you are really inferring that the name is `RequestedStatus`, so maybe something like `RequestedState` would make sense?

Reply:
Which would make the field name be `State`.

Reply:
Interesting @coreydaley ..... the name is "simple" like the analogous field of `status` employed in the tekton TaskRun, but by not using `status` it technically avoids the confusing multiple uses of `status` that the UX folks @SaschaSchwarze0 had talked to.

I'm amenable.

@SaschaSchwarze0 - any opinions from you and your folks?

I'll hold off on switching immediately, but am leaning that way.

Reply:
I would change the name of the type to `BuildRunSpecState` to go along with the prior patterns here.

Reply:
Let's move forward with anything that is not `status`. ;-) There are chances to rename this on the path towards v1.

Reply:
works for me @SaschaSchwarze0  @coreydaley :-)

will change the field name to `state`

will go with `BuildRunRequestedState` for type (change my mind since last post and went with @coreydaley 's idea)